mne/viz/epochs.py

```diff
@@ -428,27 +426,47 @@ def _prepare_epochs_image_im_data(epochs, ch_type, overlay_times, order,
         if overlay_times is not None:
             overlay_times = overlay_times[order]
 
-    if sigma > 0.:
-        data = ndimage.gaussian_filter1d(data, sigma=sigma, axis=0)
+    return data, overlay_times
 
-    # setup lims and cmap
+
+def _set_image_lims_and_scale(data, scaling=1, vmin=None, vmax=None):
+    """Set up vlims for an image (2D or 3D). Helper for plot_epochs_image."""
     scale_vmin = True if (vmin is None or callable(vmin)) else False
     scale_vmax = True if (vmax is None or callable(vmax)) else False
     vmin, vmax = _setup_vmin_vmax(
         data, vmin, vmax, norm=(data.min() >= 0) and (vmin is None))
-    if not scale_vmin:
-        vmin /= scaling
-    if not scale_vmax:
-        vmax /= scaling
+    if scale_vmin:
+        vmin *= scaling
+    if scale_vmax:
+        vmax *= scaling
+    return data * scaling, vmin, vmax
+
 
-    ylim = dict()
-    ts_args_ = dict(colors={"cond": "black"}, ylim=ylim, picks=[0], title='',
+def _prepare_tsargs_for_epochs_image(ts_args):
+    """Set default parameters for evoked plot for plot_epochs_image."""
+    ts_args_ = dict(colors={"cond": "black"}, ylim=dict(), picks=[0], title='',
                     truncate_yaxis=False, truncate_xaxis=False, show=False)
     ts_args_.update(**ts_args)
     ts_args_["vlines"] = []
+    return ts_args_
 
-    return [data * scaling, overlay_times, vmin * scaling, vmax * scaling,
-            ts_args_]
+
+def _prepare_epochs_image_im_data(data, times, ch_type, overlay_times, order,
+                                  sigma, vmin, vmax, scaling, ts_args):
+    """Preprocess epochs image (sort, filter). Helper for plot_epochs_image."""
+    from scipy import ndimage
+
+    # data transforms - sorting, scaling, smoothing
+    data, overlay_times = _order_epochs(data, times, order, overlay_times)
+
+    if sigma > 0.:
+        data = ndimage.gaussian_filter1d(data, sigma=sigma, axis=0)
```

Comment:
It is not obvious to me which axis this is smoothing. I assume 0 is still epochs, so for each channel and time point it's smoothing across epochs / repetitions (?)

Reply:
(In other words, please add a comment)

Otherwise still LGTM